src/commissaire_service/service/__init__.py

```diff
@@ -180,75 +181,89 @@ def on_message(self, body, message):
         Called when a non-jsonrpc message arrives.
 
         :param body: Body of the message.
-        :type body: str
+        :type body: dict
         :param message: The message instance.
         :type message: kombu.message.Message
         """
         self.logger.error(
             'Dropping unknown message: payload="{}", properties="{}"'.format(
                 body, message.properties))
-        message.ack()
 
     def _wrap_on_message(self, body, message):
         """
         Wraps on_message for jsonrpc routing and logging.
 
         :param body: Body of the message.
-        :type body: str
+        :type body: dict or json string
         :param message: The message instance.
         :type message: kombu.message.Message
         """
         self.logger.debug('Received message "{}" {}'.format(
             message.delivery_tag, body))
         expected_method = message.delivery_info['routing_key'].rsplit(
             '.', 1)[1]
-        # If we have a method and it matches the routing key treat it
-        # as a jsonrpc call
-        if (
-                isinstance(body, dict) and
-                'method' in body.keys() and
-                body.get('method') == expected_method):
-            try:
+
+        # If we don't get a valid message we default to -1 for the id
+        uid = -1
+        result = None
+        try:
+            # If we don't have a dict then it should be a json string
+            if not isinstance(body, dict):
+                body = json.loads(body)
+
+            # If we have a method and it matches the routing key treat it
+            # as a jsonrpc call
+            if (
+                    isinstance(body, dict) and
+                    'method' in body.keys() and
+                    body.get('method') == expected_method):
```

Comment:
nit: Since you're using `body.get('method')` here, I think you could lose the `'method' in body.keys()` part.  `expected_method` should never be `NULL`, right?  (or if it can it should be caught earlier)